src/Dev/Tasks/SecureAssetsMigrationHelper.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace SilverStripe\Assets\Dev\Tasks;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use League\Flysystem\Filesystem;
+use SilverStripe\Assets\Flysystem\FlysystemAssetStore;
+use SilverStripe\Assets\Folder;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\ORM\Queries\SQLSelect;
+
+/**
+ * Removes stray .htaccess files created through the silverstripe/secureassets module
+ * on a 3.x-based site. The 4.x protections work differently:
+ * One central assets/.htaccess file routes non-existent paths through SilverStripe,
+ * which can choose to return a file from assets/.protected.
+ * Any additional .htaccess files in folders can interfere with this logic.
+ *
+ * Note that this task does not migrate file metadata added/managed through silverstripe/secureassets.
+ * The metadata fields are the same in 4.x (File.CanViewType etc).
+ *
+ * See https://github.com/silverstripe/silverstripe-assets/issues/231
+ *
+ * @internal
+ */
+class SecureAssetsMigrationHelper
+{
+    use Injectable;
+    use Configurable;
+
+    /**
+     * @var array
+     */
+    protected $htaccessRegexes = [];
+
+    private static $dependencies = [
+        'logger' => '%$' . LoggerInterface::class,
+    ];
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct()
+    {
+        $this->logger = new NullLogger();
+
+        $this->htaccessRegexes = [
```

Comment:
I've compared the [history](https://github.com/silverstripe/silverstripe-secureassets/commits/f9d54e7b419574e583f3b958a2f21db163877971/_config.php) of the original htaccess config, and that's the only format used by secureassets